### PR TITLE
Add troubleshooting page for apps development

### PR DESCRIPTION
**Pull request by lkostrowski**
The goal is to maintain such page and extend it every time community needs help - new topic/question can be written down for the future

### CODE REVIEW

File: docs/developer/extending/apps/developing-apps/troubleshooting.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: Troubleshooting
+---
+
+Find solutions to common issues that you may face during apps development.
+
+# I see Typescript errors
```
Comment (peelar) [inline docs/developer/extending/apps/developing-apps/troubleshooting.mdx:7]:
```suggestion
# Issues
## I see Typescript errors
```

File: docs/developer/extending/apps/developing-apps/troubleshooting.mdx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Saleor [Apps](https://github.com/saleor/apps), [SDK](https://github.com/saleor/saleor-app-sdk) and [App Template](https://github.com/saleor/saleor-app-template/discussions)
+use [pnpm](https://pnpm.io/) package manager. It can be sensitive with dependencies resolutions, especially in a monorepo setup.
+
+Please ensure you have installed right version of pnpm on your machine and re-install dependencies with `pnpm install`.
```
Comment (peelar) [inline docs/developer/extending/apps/developing-apps/troubleshooting.mdx:12]:
```suggestion
Please ensure you have installed the right version of pnpm on your machine and re-install dependencies with `pnpm install`.
```

File: docs/developer/extending/apps/developing-apps/troubleshooting.mdx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+To find accurate pnpm version in your project, check root `package.json` file and find `packageManager` field.
```
Comment (peelar) [inline docs/developer/extending/apps/developing-apps/troubleshooting.mdx:14]:
```suggestion
To find the accurate pnpm version in your project, check the root `package.json` file and look for the `packageManager` field.
```

File: docs/developer/extending/apps/developing-apps/troubleshooting.mdx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+For example `"packageManager": "pnpm@8.2.0"` means that you need to install exactly `8.2.0` version of pnpm.
```
Comment (peelar) [inline docs/developer/extending/apps/developing-apps/troubleshooting.mdx:16]:
```suggestion
For example: `"packageManager": "pnpm@8.2.0"` means that you need to install exactly the `8.2.0` version of pnpm.
```

**Review by peelar — CHANGES_REQUESTED** [review]

**Review by peelar — APPROVED** [review]

File: docs/developer/extending/apps/developing-apps/troubleshooting.mdx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+## I see Typescript errors
+
+Saleor [Apps](https://github.com/saleor/apps), [SDK](https://github.com/saleor/saleor-app-sdk) and [App Template](https://github.com/saleor/saleor-app-template/discussions)
+use [pnpm](https://pnpm.io/) package manager. It can be sensitive with dependencies resolutions, especially in a monorepo setup.
+
+Please ensure you have installed the right version of pnpm on your machine and re-install dependencies with `pnpm install`.
+
+To find the accurate pnpm version in your project, check the root `package.json` file and look for the `packageManager` field.
+
+For example: `"packageManager": "pnpm@8.2.0"` means that you need to install exactly the `8.2.0` version of pnpm.
```
Comment (patrys) [inline docs/developer/extending/apps/developing-apps/troubleshooting.mdx:16]: That's good info but also something I'd like to see addressed eventually. Requiring precise micro versions of a globally-installed package requires the use of virtual environments, and I feel we need to document that if we can't have it work with a broader set of package manager releases.

**Review by patrys — APPROVED** [review]

**Pull request merged**